system/jhtree/jhtree.cpp

```diff
@@ -2668,6 +2667,7 @@ const CJHTreeNode *CNodeCache::getNode(const INodeLoader *keyIndex, unsigned iD,
             if (!ownedCacheEntry->isReady())
             {
                 const CJHTreeNode *node = keyIndex->loadNode(&fetchCycles, pos);
+                assertex(type == node->getNodeType());
```

Comment:
If we are that concerned about squeezing performance out of this code perhaps change this to an assert, or delete it.

Reply:
I think asserting here when a node is actually added is ok.  asserting when it hits the cache was less so.

Unfortunately because this function has 5 parameters the call from CKeyIndex::getNode cannot be changed into a jump which would be better.  I might revisit.